src/devtools/client/debugger/src/components/Editor/EmptyLines.tsx

```diff
@@ -41,24 +37,34 @@ export default function EmptyLines({ editor }: ELProps) {
       });
     }
 
-    function disableEmptyLines() {
-      const uniqueChunks = calculateRangeChunksForVisibleLines(editor);
+    async function disableEmptyLines() {
+      if (visibleLines === null || sourceId === null) {
+        return;
+      }
+
+      const breakpointPositions = await getBreakpointPositionsAsync(
+        replayClient,
+        sourceId,
+        visibleLines
+      );
+      const breakableLines = breakpointPositions.map(({ line }) => line);
+
+      // For performance reasons, we should only modify lines (chunks of lines in this case) once.
+      // However if the breakpoint positions change, we need to re-draw things.
+      if (memoizedDrawnLines.breakpointPositions !== breakpointPositions) {
```

Comment:
Hmm. Yes, but I think I'm probably over invalidating now that I look again, because this would also change when `visibleLines` changes and we don't actually need that. I'll relax it to be source ID Specific.

**Edit** – I remember now why I did this. It's. because this is how it was written before:
https://github.com/replayio/devtools/blob/1b9bfd91ab2ba4d4a0667c7f21e96efe574c2e99/src/devtools/client/debugger/src/components/Editor/EmptyLines.tsx#L25-L29

But I think you were also over-invalidating.